test/ruby/test_proc.rb

```diff
@@ -53,11 +53,9 @@ def self.dyna_var_check
     assert_equal(5, x)
   end
 
-  def assert_arity(n)
+  def assert_arity(n, &block)
     meta = class << self; self; end
-    b = assert_warn(/Capturing the given block using Proc\.new is deprecated/) do
-      Proc.new
-    end
+    b = Proc.new(&block)
```

Comment:
Is this copy needed here?
While the other bootstrap tests seem the tests for the copied proc, but this doesn't seem so.

Reply:
I'm not sure that this copy is needed, this was simply the easiest way to keep the tests passing.